InvenTree/templates/js/translated/tables.js

```diff
@@ -21,6 +22,62 @@ function reloadtable(table) {
 }
 
 
+/**
+ * Download data from a table, via the API.
+ * This requires a number of conditions to be met:
+ * 
+ * - The API endpoint supports data download (on the server side)
+ * - The table is "flat" (does not support multi-level loading, etc)
+ * - The table has been loaded using the inventreeTable() function, not bootstrapTable()
+ *   (Refer to the "reloadTableFilters" function to see why!)
+ */
+function downloadTableData(table, opts={}) {
+
+    // Extract table configuration options
+    var table_options = table.bootstrapTable('getOptions');
+
+    var url = table_options.url;
+
+    if (!url) {
+        console.log('Error: downloadTableData could not find "url" parameter.');
+    }
+
+    var query_params = table_options.query_params || {};
+
+    url += '?';
+
+    constructFormBody({}, {
+        title: opts.title || '{% trans "Export Table Data" %}',
+        fields: {
+            format: {
+                label: '{% trans "Format" %}',
+                help_text: '{% trans "Select File Format" %}',
+                required: true,
+                type: 'choice',
+                value: 'csv',
+                choices: exportFormatOptions(),
+            }
+        },
+        onSubmit: function(fields, form_options) {
+            var format = getFormFieldValue('format', fields['format'], form_options);
+            
+            // Hide the modal
+            $(form_options.modal).modal('hide');
+
+            for (const [key, value] of Object.entries(query_params)) {
```

Comment:
@SchrodingersGat would this make sense as a generalized helper function? Can imagine we need this on multiple places